docs/config_search.md

```diff
@@ -18,84 +18,132 @@ limitations under the License.
 
 Model Analyzer's `profile` subcommand supports multiple modes when searching to find the best model configuration.
```

Comment:
Should we have a table of contents at the top here? It feels like there are enough sections and subsections to justify it.

Reply:
I've added a ToC